### ISSUE TITLE
Repeating websockets exception

### ISSUE THREAD

**Issue by Demmenie** [tintin10q/headless-henk#37]
This error keeps occurring repeatedly, no pixels being placed:
```cmd
File "C:\Users\-----\AppData\Local\Programs\Python\Python310\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "C:\Users\-----\AppData\Local\Programs\Python\Python310\lib\threading.py", line 1371, in run
    self.function(*self.args, **self.kwargs)
  File "C:\Users\-----\github\headless-henk\client.py", line 103, in place_pixel
    self.differences = loop2.run_until_complete(
  File "C:\Users\-----\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 641, in run_until_complete
    return future.result()
  File "C:\Users\-----\github\headless-henk\images.py", line 94, in get_pixel_differences_with_canvas_download
    canvas = await build_canvas_image(canvas_indexes, username=username)
  File "C:\Users\-----\github\headless-henk\canvas.py", line 43, in build_canvas_image
    canvas_part = await get_canvas_part(image_id, username=username)
  File "C:\Users\-----\github\headless-henk\canvas.py", line 15, in get_canvas_part
    canvas_url = await reddit.get_canvas_url(canvas_id, username=username)
  File "C:\Users\-----\github\headless-henk\reddit.py", line 151, in get_canvas_url
    await websocket.recv()  # This is also ack
  File "C:\Users\-----\github\headless-henk\.venv\lib\site-packages\websockets\legacy\protocol.py", line 568, in recv
    await self.ensure_open()
  File "C:\Users\-----\github\headless-henk\.venv\lib\site-packages\websockets\legacy\protocol.py", line 935, in ensure_open
    raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedOK: received 1000 (OK) terminated; then sent 1000 (OK) terminated
```

**Comment by CarolusHaifisch** [comment]
Same issue here.

**Comment by Demmenie** [comment]
Haven't really been able to solve this issue but I got around it by using docker instead of venv and pip.

**Comment by tintin10q** [comment]
I think this is fixed in 4.0.0
